lib/compat/wordpress-6.8/post.php

```diff
@@ -1,60 +1,26 @@
 <?php
-/**
- * Temporary compatibility shims for block APIs present in Gutenberg.
- *
- * @package gutenberg
- */
 
 /**
- * Get the available rendering modes for the Block Editor.
+ * Set the default editor mode for the page post type to `template-locked`.
  *
- * post-only: This mode extracts the post blocks from the template and renders only those.
- * The idea is to allow the user to edit the post/page in isolation without the wrapping template.
+ * Note: This backports into `create_initial_post_types` in WordPress Core.
  *
- * template-locked: This mode renders both the template and the post blocks
- * but the template blocks are locked and can't be edited. The post blocks are editable.
- *
- * @return array Array of available rendering modes.
- */
-function gutenberg_post_type_rendering_modes() {
-	return array(
-		'post-only',
-		'template-locked',
-	);
-}
-
-/**
- * Add the default_rendering_mode property to the WP_Post_Type object.
- * This property can be overwritten by using the post_type_default_rendering_mode filter.
- *
- * @param array  $args      Array of post type arguments.
- * @param string $post_type Post type key.
+ * @param array $args Array of post type arguments.
  * @return array Updated array of post type arguments.
  */
-function gutenberg_post_type_default_rendering_mode( $args, $post_type ) {
-	if ( ! wp_is_block_theme() || ! current_theme_supports( 'block-templates' ) ) {
+function gutenberg_update_page_editor_support( $args ) {
+	if ( empty( $args['supports'] ) ) {
 		return $args;
 	}
 
-	// Make sure the post type supports the block editor.
-	if (
-		( isset( $args['show_in_rest'] ) && $args['show_in_rest'] ) &&
-		( ! empty( $args['supports'] ) && in_array( 'editor', $args['supports'], true ) )
-	) {
-		$rendering_mode  = 'page' === $post_type ? 'template-locked' : 'post-only';
-		$rendering_modes = gutenberg_post_type_rendering_modes();
-
-		// Validate the supplied rendering mode.
-		if (
-			isset( $args['default_rendering_mode'] ) &&
-			in_array( $args['default_rendering_mode'], $rendering_modes, true )
-		) {
-			$rendering_mode = $args['default_rendering_mode'];
-		}
-
-		$args['default_rendering_mode'] = $rendering_mode;
+	$editor_support_key = array_search( 'editor', $args['supports'], true );
+	if ( false !== $editor_support_key ) {
+		unset( $args['supports'][ $editor_support_key ] );
+		$args['supports']['editor'] = array(
+			'default_mode' => 'template-locked',
```

Comment:
I know I'm focusing on non-important issueshere, but what's the naming convention here? Should it be `default-mode` instead of `default_mode`?

I recently examined the `add_theme_support` method, which seems to use dashes instead of underscores.

Reply:
@Mamaduka I'd go with consistency while it's possible.

Reply:
Thanks, @peterwilsoncc!

I'll update the argument to use the `dash` separator.

Reply:
Here's the follow-up #68745.